src/models/devicemodel.cpp

```diff
@@ -71,20 +71,21 @@ bool DeviceModel::fromSettings(const Keys* keys) {
 
 namespace {
 
-bool sortCallback(const Device& a, const Device& b, const Keys* keys) {
-  if (a.isCurrentDevice(keys)) {
-    return true;
-  }
-
-  if (b.isCurrentDevice(keys)) {
-    return false;
-  }
-
+bool sortCallback(const Device& a, const Device& b) {
   return a.createdAt() > b.createdAt();
 }
 
 }  // anonymous namespace
 
+// Check if the current device exists, and if so move it to to the front.
+void DeviceModel::moveCurrentDevice(const Keys* keys) {
```

Comment:
To confirm, this was required for simulator to run as well?

Reply:
This provides a replacement for the code that was removed from `DeviceModel::sortCallback()` which was intended to ensure the current device is always first in the list. This was necessary to fix the static assert that I was running into and causing my simulator builds to crash. The assert may not have been fatal on device builds, but they were still occuring.